src/plugins/workspace/server/workspace_client.ts

```diff
@@ -192,6 +205,27 @@ export class WorkspaceClient implements IWorkspaceClientImpl {
           throw new Error(DUPLICATE_WORKSPACE_NAME_ERROR);
         }
       }
+
+      if (newDataSources) {
```

Comment:
Can we write unit test to cover the diff check code below? It is kind of complex.

Reply:
Sure, currently there is not any tests for workspace client, maybe we need to do some work to fulfill workspace client test mock.